Sources/PackageModel/PackageReference.swift

```diff
@@ -17,55 +18,99 @@ import TSCUtility
 /// This represents a reference to a package containing its identity and location.
 public struct PackageReference: Codable {
     /// The kind of package reference.
-    public enum Kind: String, Codable {
+    public enum Kind: Codable, Equatable {
         /// A root package.
-        case root
+        case root(AbsolutePath)
 
         /// A non-root local package.
-        case local
-
-        /// A remote package.
-        case remote
+        case fileSystem(AbsolutePath)
+
+        /// A local source package.
+        case localSourceControl(AbsolutePath)
+
+        /// A remote source package.
+        case remoteSourceControl(Foundation.URL)
```

Comment:
👀  this is the key change

Reply:
What's the difference between `fileSystem` and `localSourceControl`? Do we do anything to ensure `remoteSourceControl` is indeed remote (e.g., what if URL scheme is `file://`?), or it doesn't matter?

Reply:
Is there an opportunity for us to reuse the `SourceControl.Location` enumeration here?

https://github.com/apple/swift-package-manager/blob/51fe9f947e18fe3daccf3c2aaf6391136ba2b801/Sources/PackageModel/Manifest/PackageDependencyDescription.swift#L41-L44

Reply:
yes, my plan is to further consolidate / cleanup after this initial but big PR lands.